pytorch_lightning/accelerators/accelerator.py

```diff
@@ -76,6 +76,26 @@ def setup(self, trainer: "Trainer", model: LightningModule) -> None:
         self.setup_optimizers(trainer)
         self.connect_precision_plugin(self.precision_plugin)
 
+    def start_training(self, trainer: 'Trainer'):
+        self.training_type_plugin.start_training(trainer)
+
+    def start_testing(self, trainer: 'Trainer'):
+        self.training_type_plugin.start_testing(trainer)
+
+    def start_predicting(self, trainer: 'Trainer'):
+        self.training_type_plugin.start_predicting(trainer)
+
+    def pre_dispatch(self) -> None:
+        """Hook to do something before the training/evaluation/prediction starts."""
+        self.training_type_plugin.pre_dispatch()
+        self.precision_plugin.pre_dispatch()
+
+    def post_dispatch(self) -> None:
+        """Hook to do something before the training/evaluation/prediction starts."""
+        self.training_type_plugin.post_dispatch()
+        self.precision_plugin.post_dispatch()
+        self.teardown()
```

Comment:
Does this have to be part of post-dispatch?

Doesn't the Trainer have a teardown that should/could call this teardown?

Reply:
great point! the trainer should be responsible for calling tear down as its responsible for control over the accelerator, i'll move it

Reply:
@SeanNaren I can also understand having it here, since this is where the accelerators control ends. More a point of discussion I think. 
But to me teardown is more when you leave it finally (we should think about also calling that in the del- method)